Mathlib/Algebra/Star/Subalgebra.lean

```diff
@@ -540,7 +542,7 @@ theorem adjoin_induction' {s : Set A} {p : adjoin R s → Prop} (a : adjoin R s)
     (Hstar : ∀ x, p x → p (star x)) : p a :=
   Subtype.recOn a fun b hb => by
     refine' Exists.elim _ fun (hb : b ∈ adjoin R s) (hc : p ⟨b, hb⟩) => hc
-    apply adjoin_induction hb
+    refine adjoin_induction hb ?_ ?_ ?_ ?_ ?_
```

Comment:
Isn't the fact that this is needed a bad sign? Can we change this to an induction instead or something now?

Reply:
In theory we could, though I think it's fine to declare that out of scope for this PR. Often using the `induction` syntax, when building one induction principle out of others, is needlessly verbose.

Reply:
bors merge

The change you suggest will conflict with #11439, so I'd rather put this in as is for now.